cmd/skopeo/sync.go

```diff
@@ -240,6 +253,75 @@ func syncSourceHandler(c *cli.Context, globalDestRef types.ImageReference) (toCo
 	return
 }
 
+func syncSourceFileHandler(c *cli.Context) (toCopyList []imagesBySource, err error) {
+	var cfg sourceCfg
+	if err = cfg.Unmarshal(c.String("source-file")); err != nil {
+		return nil, fmt.Errorf("Unmarshaling failed: %s", err)
+	}
+
+	for domain, domainCfg := range cfg {
+		logrus.Infof("Processing all images for domain: %s", domain)
+
+		for imageName, tags := range domainCfg.Images {
+			var sourceReferences []types.ImageReference
+
+			sourceCtx := &types.SystemContext{
+				RegistriesDirPath:  c.GlobalString("registries.d"),
+				ArchitectureChoice: c.GlobalString("override-arch"),
+				OSChoice:           c.GlobalString("override-os"),
+				DockerCertPath:     domainCfg.CertDir,
+				// DEPRECATED: keep this here for backward compatibility, but override
+				// them if per subcommand flags are provided (see below).
+				DockerInsecureSkipTLSVerify:       !c.GlobalBoolT("tls-verify"),
+				AuthFilePath:                      c.String("authfile"),
+				DockerDaemonCertPath:              domainCfg.CertDir,
+				DockerDaemonInsecureSkipTLSVerify: !domainCfg.TlsVerify,
+			}
+			if domainCfg.TlsVerify {
+				sourceCtx.DockerInsecureSkipTLSVerify = !domainCfg.TlsVerify
```

Comment:
I think this is not needed, this is already done 2 lines above. Am I missing something?

Reply:
Those are actually two distinct fields, `DockerDaemonInsecureSkipTLSVerify` and `sourceCtx.DockerInsecureSkipTLSVerify`

Reply:
Sorry, I got confused by that.